### PR TITLE
Add markdown editor with preview option for creating TCR

### PR DESCRIPTION
**Pull request by TG1999**
Date: 8th March 2024

Developer Name: Tushar Goel

---

## Issue Ticket Number
#1162 

## Description

Replace the textbox with markdown and preview creating TCR

### Documentation Updated?

- [ ] Yes
- [ ] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [x] Yes
- [ ] No

Will update

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

</details>

Screenshot from 2024-03-11 23-19-27
Screenshot from 2024-03-11 23-18-51

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot from 2024-03-17 18-14-20

</details>

### CODE REVIEW

**Comment by prakashchoudhary07** [comment]
Tests are failing please have a look, and also please update description of the PR

**Comment by TG1999** [comment]
@prakashchoudhary07 added description and tests are fixed now, please authorize the vercel for deployment,

File: src/constants/FeatureFlags.ts [hunk 1]
```diff
@@ -1 +1,2 @@
 export const isTaskDetailsPageLinkEnabled = true;
+export const isMarkDownInTCREnabled = false;
```
Comment (iamitprakash) [inline src/constants/FeatureFlags.ts:2]: do we need this, are we using this export as of now..?

**Review by iamitprakash — CHANGES_REQUESTED** [review]
Please add test coverage stats

File: src/constants/FeatureFlags.ts (hunk 1 above)
Comment (TG1999) [reply src/constants/FeatureFlags.ts:2]: yes you are correct, not needed as of now, since it's passed as a prop, removing this. Thanks!

Comment (TG1999) [reply src/constants/FeatureFlags.ts:2]: It is used here https://github.com/Real-Dev-Squad/website-status/pull/1176/files#diff-d6eba0e51aefd6bb8791db039b409efcca2f83b2f614e4725eee7327ab6ab0aaR9

**Review by iamitprakash — APPROVED** [review]

**Comment by TG1999** [comment]
@yesyash

**Pull request merged**